[css-masonry] Update Intrinsic Sizing Rows 003 Mix2 Expected / Ref Test Case

stwrt opened the pull request:
#### b92de572696da071194dfa91f8fff456698b46ea
<pre>
[css-masonry] Update Intrinsic Sizing Rows 003 Mix2 Expected / Ref Test Case
<a href="https://bugs.webkit.org/show_bug.cgi?id=283004">https://bugs.webkit.org/show_bug.cgi?id=283004</a>
<a href="https://rdar.apple.com/problem/139740150">rdar://problem/139740150</a>

Reviewed by Sammy Gill.

* Grid 1

- Remove span for item 5, which was cuasing width issues.

* Grid 2

- Remove spans for item 5, which was causing width issues.
- The 5th item should have a height of 2ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-ref.html:

Canonical link: <a href="https://commits.webkit.org/286609@main">https://commits.webkit.org/286609@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa7b16332f010724a0a1c1b8bfc83c70d7d2174f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76479 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55514 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29385 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81006 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27755 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64656 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3807 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59968 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18079 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79546 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49883 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65678 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40296 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47278 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23167 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26078 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68404 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23499 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82452 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3855 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2540 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/68245 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4008 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65647 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67491 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16836 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11465 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9557 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3802 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3825 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7255 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5583 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-expected.html:
```diff
@@ -38,22 +38,22 @@
 <body>
 
 <section>
-<grid>
```
Why does the first column here have to be a fixed value? Why is it simply not based off the size of the content?

stwrt replied:
If you look at row 3 you will notice that the height is 1.4ch, which will cause a longer phrase like “3 3” to have a larger width in this first “column”. Since we are in a Grid here the most straightforward way to achieve the look we wanted was for the first column to be a fixed size. The “5 5” needs to line up right next to the “1” not extend far out.

sammygill replied:
Does it also work if you just remove the `grid-column: span 2` from the 5 5 so that it only spans one column? If that works that might be better imo since it introduces fewer "magic numbers"

stwrt replied:
Yes, that does work.

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-expected.html:
```diff
@@ -38,22 +38,22 @@
   <item class="hidden" style="grid-area: 4/2">0 0</item>
 </grid>
 
-<grid>
```
Why does the first column here have to be a fixed value? Why is it simply not based off the size of the content?

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-ref.html:
Same questions as the ones in the -expected

stwrt committed:
[css-masonry] Update Intrinsic Sizing Rows 003 Mix2 Expected / Ref Test Case
https://bugs.webkit.org/show_bug.cgi?id=283004
rdar://problem/139740150

Reviewed by Sammy Gill.

* Grid 1

- Remove span for item 5, which was cuasing width issues.

* Grid 2

- Remove spans for item 5, which was causing width issues.
- The 5th item should have a height of 2ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-mix2-ref.html:

Canonical link: https://commits.webkit.org/286609@main
